extensions/amp-story/0.1/amp-story-info-dialog.js

```diff
@@ -0,0 +1,261 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Action, StateProperty} from './amp-story-store-service';
+import {CSS} from '../../../build/amp-story-info-dialog-0.1.css';
+import {LocalizedStringId} from './localization';
+import {Services} from '../../../src/services';
+import {assertAbsoluteHttpOrHttpsUrl} from '../../../src/url';
+import {closest} from '../../../src/dom';
+import {createShadowRootWithStyle} from './utils';
+import {dev} from '../../../src/log';
+import {getAmpdoc} from '../../../src/service';
+import {htmlFor} from '../../../src/static-template';
+
+/** @const {string} Class to toggle the info dialog. */
+export const DIALOG_VISIBLE_CLASS = 'i-amphtml-story-info-dialog-visible';
+
+/** @const {string} Class to toggle the info dialog link. */
+export const MOREINFO_VISIBLE_CLASS = 'i-amphtml-story-info-moreinfo-visible';
+
+export class InfoDialog {
+  /**
+   * @param {!Window} win
+   * @param {!Element} parentEl Element where to append the component
+   */
+  constructor(win, parentEl) {
+    /** @private @const {!Window} */
+    this.win_ = win;
+
+    /** @private {?Element} */
+    this.element_ = null;
+
+    /** @private {?Element} */
+    this.innerContainerEl_ = null;
+
+    /** @private {boolean} */
+    this.isBuilt_ = false;
+
+    /** @private @const {!./localization.LocalizationService} */
+    this.localizationService_ = Services.localizationServiceV01(this.win_);
+
+    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = Services.storyStoreServiceV01(this.win_);
+
+    /** @private @const {!Element} */
+    this.parentEl_ = parentEl;
+
+    /** @const @private {!../../../src/service/vsync-impl.Vsync} */
+    this.vsync_ = Services.vsyncFor(this.win_);
+
+    /** @private {?Element} */
+    this.moreInfoLinkEl_ = null;
+
+    /** @const @private {!../../../src/service/viewer-impl.Viewer} */
+    this.viewer_ = Services.viewerForDoc(this.parentEl_);
+  }
+
+  /**
+   * Builds and appends the component in the story.
+   * @return {!Promise}
+   */
+  build() {
+    if (this.isBuilt()) {
+      return Promise.resolve();
+    }
+
+    this.isBuilt_ = true;
+    const root = this.win_.document.createElement('div');
+    const html = htmlFor(this.parentEl_);
+    this.element_ = html`
+        <div class="i-amphtml-story-info-dialog i-amphtml-story-system-reset">
+          <div class="i-amphtml-story-info-dialog-container">
+            <h1 class="i-amphtml-story-info-heading"></h1>
+            <a class="i-amphtml-story-info-link"></a>
+            <a class="i-amphtml-story-info-moreinfo"></a>
+          </div>
+        </div>`;
+
+    createShadowRootWithStyle(root, this.element_, CSS);
+    this.initializeListeners_();
+
+    const appendPromise = this.vsync_.runPromise({
+      mutate: () => {
+        this.innerContainerEl_ =
```

Comment:
I learned from other team members the other day that since a DOM query doesn't trigger any relayout, we don't have to execute it in a measure callback

But more importantly, it looks like you mixed the mutate/measure keywords